numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -195,7 +195,6 @@ PyArray_BufferConverter(PyObject *obj, PyArray_Chunk *buf)
      * sticks around after the release.
      */
     PyBuffer_Release(&view);
-    _dealloc_cached_buffer_info(obj);
```

Comment:
Does this mean that for arrays this is no longer dealloc'd?

Reply:
Ah, https://github.com/numpy/numpy/pull/16389/files#r430751527 discusses this.

Reply:
Yes, although they cannot reach here, I think. So maybe for void scalars, hmmm.  I am not sure its really worth it to release the buffer for void scalars here? It will get released eventually, but just adds weird overhead in the vast majority of cases where the input is not a void scalar/numpy array.

Ideally, it would also save a few cycles if the arrays buffer is exported twice, but that is fairly unfeasible right now.